packages/components/src/date-time/time.js

```diff
@@ -89,8 +100,46 @@ class TimePicker extends Component {
 		}
 		const newDate = date.clone().minutes( value );
 		this.setState( { date: newDate } );
-		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
-		onChange( formattedDate );
+		onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+	}
+
+	updateDay() {
+		const { onChange } = this.props;
+		const { day, date } = this.state;
+		const value = parseInt( day, 10 );
```

Comment:
There's a fair bit of code duplication across these functions and seems like something we could factor out to a utility function(s).

(not blocking)

Reply:
I thought that too looking at them (I might've even left a comment on this PR when I reviewed it awhile ago to that effect), but I think there's just enough that abstracting it will make things less/as clear with more actual lines of code. 🤷‍♂️